src/predicate.ts

```diff
@@ -32,9 +32,7 @@ export function isSelectionPredicate(predicate: LogicalOperand<Predicate>): pred
   return predicate && predicate['selection'];
 }
 
-export interface FieldEqualPredicate {
-  // TODO: support aggregate
-
+export interface FieldPredicateBase {
```

Comment:
This is a nice refactor, but seems irrelvant to this PR.

Reply:
I had to use it at some point but then it ended up being automatically inferred by Typescript.